src/common/store/handshake.test.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * @jest-environment node
+ */
+
+/* eslint-disable jest/expect-expect */
+
+import { encrypt } from 'eciesjs';
+import { expectSaga } from 'redux-saga-test-plan';
+
+import { createHandshakeKeyPair } from '@common/utils';
+import { fEncryptionPrivateKey, fEncryptionPublicKey } from '@fixtures';
+
+import slice, {
+  createKeyPair,
+  createKeyPairWorker,
+  postHandshake,
+  putJson,
+  sendPublicKey,
+  setHandshaken,
+  setKeyPair,
+  setPublicKeyWorker,
+  setTargetPublicKey
+} from './handshake';
+
+describe('Handshake', () => {
+  describe('setHandshaken', () => {
+    it('sets isHandshaken to the payload', () => {
+      expect(slice.reducer({ isHandshaken: false }, setHandshaken(true))).toStrictEqual({
+        isHandshaken: true
+      });
+
+      expect(slice.reducer({ isHandshaken: true }, setHandshaken(false))).toStrictEqual({
+        isHandshaken: false
+      });
+    });
+  });
+
+  describe('setKeyPair', () => {
+    it('sets publicKey and privateKey to the payload', async () => {
+      const keyPair = await createHandshakeKeyPair();
+      expect(slice.reducer({ isHandshaken: false }, setKeyPair(keyPair))).toStrictEqual(
+        expect.objectContaining({
+          ...keyPair
+        })
+      );
+    });
+  });
+
+  describe('setTargetPublicKey', () => {
+    it('sets targetPublicKey to the payload', () => {
+      expect(
+        slice.reducer({ isHandshaken: false }, setTargetPublicKey(fEncryptionPublicKey))
+      ).toStrictEqual(
+        expect.objectContaining({
+          targetPublicKey: fEncryptionPublicKey
+        })
+      );
+    });
+  });
+});
+
+describe('putJson', () => {
+  it('puts an action if allowInsecure is set or if the action is handshake/sendPublicKey', async () => {
+    const action = JSON.stringify(sendPublicKey(fEncryptionPublicKey));
+    await expectSaga(putJson, action)
+      .put({ ...sendPublicKey(fEncryptionPublicKey), remote: true })
+      .silentRun();
+
+    const insecureAction = JSON.stringify(setHandshaken(true));
+    await expectSaga(putJson, insecureAction, true)
+      .put({ ...setHandshaken(true), remote: true })
+      .silentRun();
+  });
+
+  it('decrypts an encrypted action', async () => {
+    const insecureAction = JSON.stringify(setHandshaken(true));
```

Comment:
Can we test with something that is supposed to be synced? This is a bit asking for trouble 😆

Reply:
Same goes for the other tests in this file